Accessibility crash

‎`‎java.lang.RuntimeException: Unable to stop service com.eblan.launcher.service.EblanAccessibilityService@dcb92ab: java.lang.IllegalArgumentException: Receiver not registered: com.eblan.launcher.service.EblanAccessibilityService$globalActionBroadcastReceiver$1@c3106bb
‎at android.app.ActivityThread.handleStopService(ActivityThread.java:4886)
‎at android.app.ActivityThread.access$2300(ActivityThread.java:272)
‎at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2221)
‎at android.os.Handler.dispatchMessage(Handler.java:106)
‎at android.os.Looper.loopOnce(Looper.java:241)
‎at android.os.Looper.loop(Looper.java:342)
‎at android.app.ActivityThread.main(ActivityThread.java:8118)
‎at java.lang.reflect.Method.invoke(Native Method)
‎at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:583)
‎at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1045)
‎Caused by: java.lang.IllegalArgumentException: Receiver not registered: com.eblan.launcher.service.EblanAccessibilityService$globalActionBroadcastReceiver$1@c3106bb
‎at android.app.LoadedApk.forgetReceiverDispatcher(LoadedApk.java:1573)
‎at android.app.ContextImpl.unregisterReceiver(ContextImpl.java:1891)
‎at android.content.ContextWrapper.unregisterReceiver(ContextWrapper.java:786)
‎at com.eblan.launcher.service.EblanAccessibilityService.onDestroy(EblanAccessibilityService.kt:77)
‎at android.app.ActivityThread.handleStopService(ActivityThread.java:4866)
‎... 9 more`

When stopping the AccessibilityService
‎

--- a/service/src/main/kotlin/com/eblan/launcher/service/EblanAccessibilityService.kt
+++ b/service/src/main/kotlin/com/eblan/launcher/service/EblanAccessibilityService.kt
@@ -63,12 +63,10 @@ internal class EblanAccessibilityService : AccessibilityService() {
     }
 
     override fun onServiceConnected() {
-        val intentFilter = IntentFilter(GlobalAction.NAME)
-
         ContextCompat.registerReceiver(
-            applicationContext,
+            this,
             globalActionBroadcastReceiver,
-            intentFilter,
+            IntentFilter(GlobalAction.NAME),
             ContextCompat.RECEIVER_NOT_EXPORTED,
         )
     }

